### PR TITLE
Add tests for NH-3755 - Proxy exception for multiple joined-subclass

### PR DESCRIPTION
**Pull request by ynauls**
Created unit test to reproduce exception after swapping proxy factory due to upgrade from NH 3.2 to 4.0 
ProxyFactory:
from _NHibernate.ByteCode.Castle.ProxyFactoryFactory, NHibernate.ByteCode.Castle_
to _NHibernate.Bytecode.DefaultProxyFactoryFactory, NHibernate_

https://nhibernate.jira.com/browse/NH-3755

### CODE REVIEW

File: src/NHibernate.Test/NHSpecificTest/NH3755/Fixture.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				session.Delete("from System.Object");
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+        [Ignore]
```
Comment (hazzik) [inline src/NHibernate.Test/NHSpecificTest/NH3755/Fixture.cs]: Can you please apply [KnownBug] attribute?

File: src/NHibernate.Test/NHSpecificTest/NH3755/FixtureByCode.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+			using (ITransaction transaction = session.BeginTransaction())
+			{
+				session.Delete("from System.Object");
+
+				session.Flush();
+				transaction.Commit();
+			}
+		}
+
+		[Test]
```
Comment (hazzik) [inline src/NHibernate.Test/NHSpecificTest/NH3755/FixtureByCode.cs]: Can you please apply [KnownBug]

File: src/NHibernate.Test/NHSpecificTest/NH3755/Fixture.cs (hunk 1 above)
Comment (piolo108) [inline src/NHibernate.Test/NHSpecificTest/NH3755/Fixture.cs]: is this attribute in addition or replace?  Also, if its a known issue, is there a timeline for the fix? I can assist

Comment (hazzik) [inline src/NHibernate.Test/NHSpecificTest/NH3755/Fixture.cs]: To replace, the idea of [KnownBug] that the test will still run, and we expect it to fail. And... as soon as you make an issue in JIRA it become a known bug:)

Comment (piolo108) [inline src/NHibernate.Test/NHSpecificTest/NH3755/Fixture.cs]: attribute has been applied

**Comment by hazzik** [comment]
@piolo108 can you please squash the commits and update the pull request?

**Comment by piolo108** [comment]
@hazzik please excuse my inexperience with github, but what does that exactly mean :) or how do i accomplish that?

**Comment by hazzik** [comment]
Something like this http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

Then you push to github with --force flag.

**Comment by piolo108** [comment]
@hazzik, okay, so i squashed the commits into "https://github.com/piolo108/nhibernate-core/commit/517f9239aceef16d74936484f6537f6d752cb16f", but i think i was not supposed to merge.  Is this where your is this where the force push should have come into play?

**Comment by hazzik** [comment]
>Is this where force push should have come into play?

Yes

>If its easier i can fork and create a new pull request.

Please no, as PR here are often rebased to reflect some changes

**Comment by ynauls** [comment]
@hazzik, are there steps to ratify my latest commit or are we good to merge?

**Comment by hazzik** [comment]
@ynauls should be:

> git checkout NH-3755
> git reset 517f9239aceef16d74936484f6537f6d752cb16f --hard
> git push origin NH-3755:NH-3755 -f

**Comment by ynauls** [comment]
Thank you for the assistance @hazzik, its done

**Comment by ynauls** [comment]
@hazzik, please let me know if there is anything else for me to do.

**Comment by hazzik** [comment]
Can you please enable [Allow edits from maintainers](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) PR option?

**Comment by hazzik** [comment]
Replaced by #608

**Pull request closed without merging**